### PR TITLE
modified export csv handling

### PR DESCRIPTION
**Pull request by cabch**
- used .join() + .map()
- put double quote and escaped double quotes in text to keep commas inside text
- handled null headers to still have CRLF
- for charging stations, logs, transactions, statistics and users

### CODE REVIEW

File: src/server/rest/v1/service/ChargingStationService.ts [hunk 1]
```diff
@@ -1525,80 +1525,102 @@ export default class ChargingStationService {
...
+        'vendor',
+        'firmwareVersion',
+        'ocppVersion',
+        'ocppProtocol',
+        'lastSeen',
+        'lastReboot',
+        'maxPower',
+        'powerLimitUnit'
+      ];
+      headers = headerArray.join(Constants.CSV_SEPARATOR);
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/ChargingStationService.ts:1581]: As dicussed, join can be applied right away in the array

File: src/server/rest/v1/service/ChargingStationService.ts [hunk 2]
```diff
@@ -1525,80 +1525,102 @@ export default class ChargingStationService {
...
+        chargingStation.chargePointVendor,
+        chargingStation.firmwareVersion,
+        chargingStation.ocppVersion,
+        chargingStation.ocppProtocol,
+        i18nManager.formatDateTime(chargingStation.lastSeen, 'L') + ' ' + i18nManager.formatDateTime(chargingStation.lastSeen, 'LT'),
+        i18nManager.formatDateTime(chargingStation.lastReboot, 'L') + ' ' + i18nManager.formatDateTime(chargingStation.lastReboot, 'LT'),
+        chargingStation.maximumPower,
+        chargingStation.powerLimitUnit
+      ].map((value) => {
+        return typeof value === 'string' ? '"' + value.replace('"', '""') + '"': value;
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/ChargingStationService.ts:1603]: Add space before ':'

File: src/server/rest/v1/service/LoggingService.ts [hunk 3]
```diff
@@ -55,36 +55,43 @@ export default class LoggingService {
...
+        'level',
+        'type',
+        'action',
+        'message',
+        'method',
+        'module',
+        'source',
+        'host',
+        'process'
+      ]
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/LoggingService.ts:73]: Why don't you use the same approach as in Charging Station and provide the join() here?

File: src/server/rest/v1/service/TransactionService.ts [hunk 4]
```diff
@@ -889,48 +889,56 @@ export default class TransactionService {
...
+        'startTime',
+        'endDate',
+        'endTime',
+        'totalConsumption',
+        'totalDuration',
+        'totalInactivity',
+        'price',
+        'priceUnit'
+      ];
+      headers = headerArray.join(Constants.CSV_SEPARATOR);
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/TransactionService.ts:914]: Same remark as in Charging Station service

File: src/server/rest/v1/service/UserService.ts [hunk 5]
```diff
@@ -1338,36 +1338,44 @@ export default class UserService {
...
+        'locale',
+        'role',
+        'status',
+        'email',
+        'eulaAcceptedOn',
+        'createdOn',
+        'changedOn',
+        'changedBy'
+      ];
+      headers = headerArray.join(Constants.CSV_SEPARATOR);
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/UserService.ts:1357]: Same remark as in Charging Station service

**Review by LucasBrazi06 — APPROVED** [review]

**Review by LucasBrazi06 — APPROVED** [review]
Check my comments

File: src/server/rest/v1/service/LoggingService.ts (hunk 3 above)
Comment (cabch) [reply src/server/rest/v1/service/LoggingService.ts:73]: when headers are only strings (no computing in the array) we can skip the join. Anyways I re-added it already to be consistent with other parts and not having surprise if one day we need changes

**Pull request merged**